### PR TITLE
Closes #49 - Add preset setting to timer page

### LINKED ISSUE #49: Add a presets setting to Timer page
Add to user a chance to work with timers presets. Create a presets settings menu where user can create his custom preset with so many timers as he wants. After this , add a possibility to update preset: rename preset name or rename timers name and update timer fields , delete old timers and create new. Add a possibility to delete a custom preset. For all unregistered users , add a chance to use a standard preset. Also provide a visual feedback  about the preset: preset name , list of timers in preset and current timer.

### PR DESCRIPTION
**Pull request by YurasNazar**

### CODE REVIEW

File: src/app/services/preset.service.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    return this.http.post(this.config.urlPreset + 'CreatePreset', presetModelJson, { headers: headers });
+  }
+
+  getGetAllStandardPresetsFromServer(): Observable<PresetModelJson[]> {
+    const headers = this.getHeaders();
+    return this.http.get<PresetModelJson[]>(this.config.urlPreset + 'GetAllStandardPresets', { headers: headers });
+  }
+
+  getGetAllCustomPresetsFromServer(): Observable<PresetModelJson[]> {
+    const headers = this.getHeaders();
```
Comment (roman-fay) [inline src/app/services/preset.service.ts]: as an option, you can simply put this.getHeaders directly to headers:

Comment (YurasNazar) [reply src/app/services/preset.service.ts]: Okey. In the next commit I will fix.

File: src/app/services/preset.service.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { PresetModel, Timer, PresetModelJson, TimerJson } from '../models/preset.model';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { ConfigService } from './config.service';
+import { TimerService } from './timer.service';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+
```
[the remark this answers is not in the document]
Comment (YurasNazar) [reply src/app/services/preset.service.ts]: Thank you :). I will fix.

File: src/app/preset/preset.component.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  isLoggedIn: boolean;
+  
+  constructor(
+    private presetsFormDialogRef: MatDialogRef<PresetComponent>,
+    private signupDialogComponent: SignupDialogComponent,
+    private presetService: PresetService,
+    private timerService:  TimerService,
+    private formBuilder:   FormBuilder){}
+ 
+
```
[the remark this answers is not in the document]
Comment (YurasNazar) [reply src/app/preset/preset.component.ts]: Thank you. I will fix.

File: src/app/preset/preset.component.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
+import { SignupDialogComponent } from '../signup-dialog/signup-dialog.component';
+import { FormGroup, FormBuilder, Validators , FormArray } from '@angular/forms';
+import { PresetService } from '../services/preset.service';
+import { TimerService } from '../services/timer.service';
+import { PresetModel } from '../models/preset.model';
+import { Component, OnInit } from '@angular/core';
+import { MatDialogRef } from '@angular/material';
+
+
```
[the remark this answers is not in the document]
Comment (YurasNazar) [reply src/app/preset/preset.component.ts]: Ok , in the next commit I will fix it.

**Pull request merged**